core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -3185,16 +3197,101 @@ class DAGSchedulerSuite extends SparkFunSuite with TempLocalSparkContext with Ti
       "Spark can only do this while using the new shuffle block fetching protocol"))
   }
 
+  test("SPARK-51272: retry all the succeeding stages when the map stage is indeterminate with" +
+    " concurrent tasks completion") {
+    if (scheduler != null) {
+      this.afterEach()
+    }
+    val resubmitFailedStageTriggered = Array.fill[Boolean](1)(false)
```

Comment:
Why is this not a boolean directly? Why are we using an array with only one element? Can't we have a single variable?

Reply:
Let me see if I can remove it.. It is old habit from Java, where anonymous inner class can  access only final fields of outer class.
I am not sure if it holds true for scala.. I mean if the anonymous inner class in scala can access mutable fields of outer class and mutate them, then array can be replaced by a single var boolean

Reply:
done. thanks @sririshindra  for pointing it out.